proposals/nnnn-video-streaming-state.md

```diff
@@ -0,0 +1,165 @@
+</enum>
+```
+
+#### HMI PI
+This proposal also add a new enum value `PROJECTION` to data type `EventTypes`.
+HMI shall send a `OnEventChanged` notification with `EventTypes`=`PROJECTION` to SDL when HMI brings a projection app to foreground (`isActive`=`true`) or HMI switches from the foreground projection app (`isActive`=`false`).
+
+
+```xml
+<enum name="EventTypes">
+  <description>Reflects the current active event</description>
+  <element name="AUDIO_SOURCE">
+    <description>Navigated to audio(radio, etc)</description>
+  </element>
+  <element name="EMBEDDED_NAVI">
+    <description>Navigated to navigation screen</description>
+  </element>
+  <element name="PHONE_CALL">
+    <description>Phone call is active</description>
+  </element>
+  <element name="EMERGENCY_EVENT">
+    <description>Active emergency event, active parking event</description>
+  </element>
+  <element name="DEACTIVATE_HMI">
+    <description> GAL/DIO is active </description>
+  </element>
+  
+  <!-- new additions-->
+  <element name="PROJECTION">
+    <description>Projection app is shown in HMI</description>
+  </element>
+</enum>
+```
+
+#### Compatibility
+- New apps on old HUs: Old HUs do not support new projection type app. Since old HUs does not send the new videoStreamingState parameter, on mobile proxy, we can give default value as *streamable* to  the parameter and make sure app does not do streaming in `BACKGROUND` or `NONE`  to make new navigation apps work with old HUs.
+- Old apps on new HUs: the old navigation app does not recognize the new parameter, thus ignores it. Since there are navigation apps only (no old projection apps), the app use HMI level as an indication to stop video streaming as it current does. If a new non-media projection app is launched, the old Nav app moves to LIMITED and continues to stream its video. SDL shall send stop service control frame to the app.
+
+
+## Potential downsides
+On iOS devices, we still lack an effective way to bring a background Navigation/Projection app to foreground so that the app can restart/resume video streaming when the driver brings the app back to `FULL` in HMI.
+
+
+## Impact on existing code
+
+- RPC need get updated with new parameter. 
+- SDL core and mobile proxy lib need updates to support this new parameter. 
+- App's HMI level transition logic in SDL core need updates.
+
+
+## Alternatives considered
+
+- Instead of using a new enum data type `VideoStreamingState`, we can use just `Boolean` type and rename `videoStreamingState` to `isVideoSource` as the following.
+
+```xml
+<function name="OnHMIStatus" functionID="OnHMIStatusID" messagetype="notification">
+    ...
+    
+    <!-- new additions-->
+    <param name="isVideoSource" type="Boolean" mandatory="true">
+        <description>Indicates whehter an app can use video service or not.
+        If it is true, the mobile app can use video service.
+        If it is false, the mobile app must stop streaming video to SDL.        
```

Comment:
Should this be updated to "...streaming video sto SDL Core." ?

Reply:
changed